src/Entity/ItemId.php

```diff
@@ -20,29 +20,31 @@ class ItemId extends EntityId implements Int32EntityId {
 
 	/**
 	 * @param string $idSerialization
+	 * @param string $repositoryName
 	 *
 	 * @throws InvalidArgumentException
 	 */
-	public function __construct( $idSerialization ) {
-		$serializationParts = self::splitSerialization( $idSerialization );
-		$localId = strtoupper( $serializationParts[2] );
-		$this->assertValidIdFormat( $localId );
-		parent::__construct( self::joinSerialization(
-			array( $serializationParts[0], $serializationParts[1], $localId ) )
-		);
-	}
-
-	private function assertValidIdFormat( $idSerialization ) {
+	public function __construct( $idSerialization, $repositoryName = '' ) {
 		if ( !is_string( $idSerialization ) ) {
 			throw new InvalidArgumentException( '$idSerialization must be a string' );
 		}
 
-		if ( !preg_match( self::PATTERN, $idSerialization ) ) {
+		$parts = explode( ':', $idSerialization );
```

Comment:
Having the explode/implode based on ":" here means spreading the knowledge about the syntax for the prefixes across multiple classes. That's what I tried to avoid with the static methods.